src/common/serializer/buffered_file_writer.cpp

```diff
@@ -22,16 +23,32 @@ idx_t BufferedFileWriter::GetTotalWritten() {
 }
 
 void BufferedFileWriter::WriteData(const_data_ptr_t buffer, idx_t write_size) {
-	// first copy anything we can from the buffer
-	const_data_ptr_t end_ptr = buffer + write_size;
-	while (buffer < end_ptr) {
-		idx_t to_write = MinValue<idx_t>((end_ptr - buffer), FILE_BUFFER_SIZE - offset);
-		D_ASSERT(to_write > 0);
-		memcpy(data.get() + offset, buffer, to_write);
-		offset += to_write;
-		buffer += to_write;
-		if (offset == FILE_BUFFER_SIZE) {
-			Flush();
+	if (write_size >= (FILE_BUFFER_SIZE + offset)) {
+		// Perform direct IO, the buffer to write is larger than the internal buffer
+		// it does not make sens to split a larger buffer into smaller one
+		idx_t to_copy = 0;
+		if (offset != 0) {
```

Comment:
Can this comment be clarified further?
It sounds like we're not appending to our buffer and flushing it as is, but in reality we're filling our buffer up to FILE_BUFFER_SIZE and then flushing it.

Reply:
is it better ? [3931b0b](https://github.com/duckdb/duckdb/pull/11203/commits/3931b0bb5aa72c0a6d0bf7e30bca6a921f9e3a2b)